### PR TITLE
Port mrpPD module to C++

### LINKED ISSUE #195: Port mrpPD module to C++
**Describe your use case**
The mrpPD module will be ported to C++. The old module name is unchanged; while the new module is called `mrpProportionalDerivative`.

**Describe alternatives solutions you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by leahkiner**
* **Tickets addressed:** bsk-750
* **Review:** By commit
* **Merge strategy:** Merge (no squash) 

## Description
This PR creates a C++ version of the `mprPD` fsw module.

## Verification
The old unit test script is kept and the  new module passes the test.

## Documentation
N/A

## Future work
N/A

### CODE REVIEW

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: :large_orange_diamond: In a separate commit we should check all error meessages and update them.

You can also remove the `Error: ` from the start of the messages as bskLog already prepends an indication of error.

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: :small_red_triangle_down: this variable is only relevant within the if statement. To ensure variables are given the smallest scope necessary, let's declare it within the if statement and there's no need to call the `vcInMsg = VehicleConfigMsgPayload();`, if I'm not mistaken.

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: :small_red_triangle_down: `macroDefinitions.h` and `string.h` are not used and can be removed.

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: :small_red_triangle_down: let's name this variable different to the one that is currently defined as a member of the class. Maybe `guidanceMsgPayload` or something to indicate it's not a message but rather the payload.

Same for `controlOutMsg`

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: :small_red_triangle_down: A `const` declaration has no effect on types like double, float, int etc. This can be removed.

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: :small_red_triangle_down: Same here

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/test_mrpProportionalDerivative.py
Comment (patkenneally) [inline src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/test_mrpProportionalDerivative.py]: :small_red_triangle_down: Let's refactor the test (in a later commit) to no longer use the testFailCount pattern. Rather just assertions (np.testing etc.) and no calling a first function to only then call a second test function.

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/Support/results_MRP_PD.py
Comment (patkenneally) [inline src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/Support/results_MRP_PD.py]: Is this file needed? It doesn't look like it's used for anything.

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/test_mrpProportionalDerivative.py
Comment (patkenneally) [inline src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/test_mrpProportionalDerivative.py]: 🔶 The author in all these module's files can be adjusted to "Laboratory for Atmospheric and Space Physics, University of Colorado at Boulder"

**Review by patkenneally — CHANGES_REQUESTED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

Nice work

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/Support/results_MRP_PD.py
Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/Support/results_MRP_PD.py]: I don't think so, I added all of the files in the mrpPD folder to this PR

Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/Support/results_MRP_PD.py]: I just removed the file

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/test_mrpProportionalDerivative.py
Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/test_mrpProportionalDerivative.py]: Done

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: Done

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: Done

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: Done. I renamed `guidInMsg` to `guidanceMsgPayload` and `controlOutMsg` to `torqueCmdMsgPayload`.

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: Done

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: Done

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/test_mrpProportionalDerivative.py
Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/_UnitTest/test_mrpProportionalDerivative.py]: Done

File: src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attControl/mrpProportionalDerivative/mrpProportionalDerivative.cpp]: Done

**Review by patkenneally — APPROVED** [review]

**Pull request merged**
